src/borg/platform/base.py

```diff
@@ -283,3 +285,22 @@ def process_alive(host, pid, thread):
 def local_pid_alive(pid):
     """Return whether *pid* is alive."""
     raise NotImplementedError
+
+
+def get_free_space(path):
+    if os.name != 'nt':
+        try:
+            st_vfs = os.statvfs(path)
+        except OSError as os_error:
+            logger.warning('Failed to check free space before committing: ' + str(os_error))
+            return
+        except AttributeError:
+            # TODO move the call to statvfs to platform
+            logger.warning('Failed to check free space before committing: no statvfs method available')
+            return
+        # f_bavail: even as root - don't touch the Federal Block Reserve!
+        free_space = st_vfs.f_bavail * st_vfs.f_frsize
+    else:
+        import psutil
+        free_space = psutil.disk_usage(path).free
+    return free_space
```

Comment:
as this is platform dependent, rather not put it into `base`, but into the os platform modules?

Reply:
Would it be possible to add psutils as a dependency? Because this would remove some of the platform specific code.  For example `get_free_space` and `process_alive` would no longer require a platform dependet implementation. Maybe there are other functions too.

Reply:
shutil.disk_usage() seems to work for disk usage, It is listed as Unix and Windows.

Reply:
if you can make it work without psutils (and without much work), that would be preferable.